packages/runtime/container-runtime/src/containerRuntime.ts

```diff
@@ -4366,7 +4368,9 @@ export class ContainerRuntime
 	}
 
 	private reSubmit(message: PendingMessageResubmitData): void {
-		// Need to parse from string for back-compat
+		// Messages in the PendingStateManager and Outbox (both call resubmit) have serialized contents, so parse it here.
+		// Note that roundtripping handles through string like this means this parsed contents
+		// contains RemoteFluidObjectHandles, not the original handle.
```

Comment:
this is wrong if true, the plan requires it be the real handle, not the remote object handle, as the remote object handle doesn't know how to bind the object.

Reply:
i see that the plan is to move away from this in the pr comments, so probably ok for now

Reply:
Yeah, separating that to the next PR